composeApp/src/commonMain/kotlin/com/github/familyvault/services/FileCabinetService.kt

```diff
@@ -36,71 +46,270 @@ class FileCabinetService(
             managers = managers,
             tag = AppConfig.FILE_CABINET_THREAD_TAG,
             type = FileCabinetThreadType.IMAGES.toString(),
-            name = familyGroupName,
-            referenceStoreId = storeId,
+            name = "$familyGroupName$IMAGES_SUFFIX",
+            referenceStoreId = imagesStoreId,
+            threadInitialCreators = emptyList()
+        )
+
+        val documentsStoreId = privMxClient.createStore(
+            contextId,
+            users,
+            managers,
+            StoreType.FILE_CABINET_DOCUMENTS.toString()
+        )
+        
+        privMxClient.createThread(
+            contextId = contextId,
+            users = users,
+            managers = managers,
+            tag = AppConfig.FILE_CABINET_THREAD_TAG,
+            type = FileCabinetThreadType.DOCUMENTS.toString(),
+            name = "$familyGroupName$DOCUMENTS_SUFFIX",
+            referenceStoreId = documentsStoreId,
             threadInitialCreators = emptyList()
         )
     }
 
     override fun retrieveFileCabinetThread(): ThreadItem {
+        return retrieveFileCabinetImagesThread()
+    }
+
+    override fun retrieveFileCabinetImagesThread(): ThreadItem {
+        val contextId = familyGroupSessionService.getContextId()
+        val fileCabinetThreads = privMxClient.retrieveAllThreadsWithTag(
+            contextId,
+            AppConfig.FILE_CABINET_THREAD_TAG,
+            startIndex = 0,
+            pageSize = 100
+        )
+
+        val imagesThread = fileCabinetThreads.firstOrNull { 
+            it.privateMeta.name.endsWith(IMAGES_SUFFIX)
+        }
+
+        return imagesThread ?: fileCabinetThreads.firstOrNull { thread ->
+            !thread.privateMeta.name.endsWith(DOCUMENTS_SUFFIX)
+        } ?: throw IllegalStateException("No file cabinet images thread found")
```

Comment:
Rozumiem, że jeżeli nie znajdziemy thread który kończy się na ` - Images` to bierzemy pierwszy lepszy thread który nie jest thread z `- Documents`? Ja bym to pominął i od razu wyrzucił wyjątek, nie mamy żadnej pewności że to będzie thread dla obrazów.